src/main/java/analysis/GoblintAnalysis.java

```diff
@@ -111,24 +115,28 @@ private void preAnalyse() {
 
 
     /**
-     * Sends the request to Goblint server to reanalyse.
+     * Sends the request to Goblint server to reanalyse and reads the result.
      *
      * @return returns true if the request was sucessful, false otherwise
      */
 
     private boolean reanalyse() {
 
         // {"jsonrpc":"2.0","id":0,"method":"analyze","params":{}}
-        String request = new GsonBuilder().create().toJson(new Request("analyze"));
+        String request1 = new GsonBuilder().create().toJson(new Request("analyze")) + "\n";
```

Comment:
So I spent 2 hours on this Friday evening, thinking why the Inputstream gets stuck... Tried a ton of stuff until Vesal came to help, after 2 additional hours of thinking, he came up with the idea to put an end-of-line to the end of the string of the JSON. That resolved the problem. 

Is this reasonable, though? Maybe we should do something on the Goblint server side so that we could pass only the JSON without this extra step? Or maybe I should use some library for that? What do you think, @sim642?

Reply:
The newline is indeed necessary right now, because the server mode reads JSON line-by-line: https://github.com/goblint/analyzer/blob/4c5135629f6f5456dcd2b57a4458c1dea985a2b0/src/util/server.ml#L73.

But that's an artificial requirement indeed and my changes in https://github.com/goblint/analyzer/pull/599 change it. The extra newline wouldn't hurt then either, so this should be good.